test/recipes/25-test_req.t

```diff
@@ -590,3 +590,30 @@ $cert = "self-signed_CA_with_keyUsages.pem";
 generate_cert($cert, "-in", srctop_file(@certs, "ext-check.csr"),
     "-copy_extensions", "copy");
 has_keyUsage($cert, 1);
+
+subtest "Issue 21403 - '-modulus' interaction with 'req'" => sub {
+    plan tests => 2;
+
+    SKIP: {
+        skip "RSA is not supported by this OpenSSL build", 2
+            if disabled("rsa");
+
+        # Tests for issue #21403 - within isolated variable scope
+        my $i21403_cnf = srctop_file('test', 'test.cnf');
+        my $i21403_key = srctop_file('test-runs', 'key-21403.pem');
+        my $i21403_cert = srctop_file('test-runs', 'cert-21403.pem');
```

Comment:
no that the test-runs folder is in the build-tree not always in the src-tree,
I think placing those files in the current working directory is the right thing to do here.
Also please do a rebase and squash your commits, instead of merge commits.
And for minor changes please use --fixup commits.  Thanks!